scalding-core/src/main/scala/com/twitter/scalding/CascadingTokenUpdater.scala

```diff
@@ -55,8 +57,23 @@ object CascadingTokenUpdater {
 
   def update(config: Config, clazzes: Set[Class[_]]): Config = {
     val toks = config.getCascadingSerializationTokens
+
+    val serializations = config.get(Config.IoSerializationsKey).getOrElse("")
+    val fromSerializations: Seq[String] = if (serializations.isEmpty)
+      Seq.empty
+    else
+      for {
+        serialization <- serializations.split(",")
+        clazz = Class.forName(serialization)
+        tokenAnnotation = clazz.getAnnotation(classOf[SerializationToken])
+        if tokenAnnotation != null
+        className <- tokenAnnotation.classNames()
+      } yield {
+        className
+      }
+
     // We don't want to assign tokens to classes already in the map
-    val newClasses: Iterable[String] = clazzes.map { _.getName } -- toks.values
+    val newClasses: Iterable[String] = clazzes.map { _.getName } -- fromSerializations -- toks.values
```

Comment:
should we be checking if the number of these tokens exceeds what cascading supports? (max int?). Wondering if there are some really complex types that end up referencing a large graph of types.